Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -13,7 +13,8 @@ class ChatClientUpdater {
 
     func prepareEnvironment(
         userInfo: UserInfo?,
-        newToken: Token
+        newToken: Token,
+        completion: @escaping (Error?) -> Void
     ) throws {
```

Comment:
Same as before. Either a throwing function or reporting an error via completion block

Reply:
it actually makes sense here, refactoring